### PR TITLE
[clang][OpenMP] Add codegen for scope directive

### PR DESCRIPTION
**Pull request by ddpagan**
Added codegen for scope directive, enabled allocate and firstprivate clauses, and added scope directive LIT test.

Testing
  - LIT tests (including new scope test).
  - OpenMP scope example test from 5.2 OpenMP API examples document.
  - Three executable scope tests from OpenMP_VV/sollve_vv suite.

### CODE REVIEW

File: clang/test/OpenMP/scope_codegen.cpp [hunk 1]
```diff
@@ -0,0 +1,2267 @@
...
+typedef void **omp_allocator_handle_t;
+extern const omp_allocator_handle_t omp_null_allocator;
+extern const omp_allocator_handle_t omp_default_mem_alloc;
+extern const omp_allocator_handle_t omp_large_cap_mem_alloc;
+extern const omp_allocator_handle_t omp_const_mem_alloc;
+extern const omp_allocator_handle_t omp_high_bw_mem_alloc;
+extern const omp_allocator_handle_t omp_low_lat_mem_alloc;
+extern const omp_allocator_handle_t omp_cgroup_mem_alloc;
+extern const omp_allocator_handle_t omp_pteam_mem_alloc;
+extern const omp_allocator_handle_t omp_thread_mem_alloc;
```
Comment (alexey-bataev) [inline clang/test/OpenMP/scope_codegen.cpp:29]: Why do you need these decls?

Comment (ddpagan) [reply clang/test/OpenMP/scope_codegen.cpp:29]: Thanks for catching this, Alexey. They aren't necessary. It was a cut/paste error as I was creating the test. I'll remove them.

Comment (ddpagan) [reply clang/test/OpenMP/scope_codegen.cpp:29]: @alexey-bataev - Actually, I forgot that they are needed by the test case as it tests using the 'allocate' clause with the scope directive.

**Review by alexey-bataev — APPROVED** [review]
LG

**Pull request merged**